### PR TITLE
Replace search in query api

### PR DESCRIPTION
**Pull request by Andrei-Dolgolev**
Replace doble search in spire API for queiries.

So instead using `bc.search` private queries return buck to `bc.get_entry`.

### list queries
```
GET http://127.0.0.1:7481/queries/list
Authorization: Bearer <moonstream-token>
```

### Create query
```
POST http://127.0.0.1:7481/queries/
Authorization: Bearer <moonstream-token>

{
    "name": "test_create_query",
    "query": "select 1"
}
```

### delete query
```
DELETE  http://127.0.0.1:7481/queries/test_create_query
Authorization: Bearer <moonstream-token>
```

### update query
```
PUT http://127.0.0.1:7481/queries/test_create_query
Authorization: Bearer <moonstream-token>

{
    "query": "select address from polygon_labels where label='moonworm-alpha' limit 1"
}
```

### test update data
```
POST http://127.0.0.1:7481/queries/test_create_query/update_data
Authorization: Bearer <moonstream-token>

{
    "params": {

    }
}
```

### test template update data
```
POST http://127.0.0.1:7481/queries/template_tokens_metadata/update_data
Authorization: Bearer <moonstream-token>

{
    "blockchain": "polygon",
    "params": {
        "address" : "0xA2a13cE1824F3916fC84C65e559391fc6674e6e8"
    }
}
```

###  test query access s3 presigned url
```
POST http://127.0.0.1:7481/queries/test_create_query
Authorization: Bearer <moonstream-token>

{
    "params": {

    }
}

```

### CODE REVIEW

File: moonstreamapi/moonstreamapi/routes/queries.py [hunk 1]
```diff
@@ -117,6 +121,7 @@ async def create_query_handler(
             title=f"Query:{query_name}",
             tags=["type:query"],
             content=query_applied.query,
+            timeout=10,
```
Comment (kompotkot) [inline moonstreamapi/moonstreamapi/routes/queries.py:124]: We have variable in settings.py `BUGOUT_REQUEST_TIMEOUT_SECONDS` what if we will use something like `BUGOUT_REQUEST_TIMEOUT_SECONDS * 2` ? At lease we will reduce hardcoded timeouts, or we can create addtional variable

File: moonstreamapi/moonstreamapi/routes/queries.py [hunk 2]
```diff
@@ -263,36 +274,34 @@ async def get_query_handler(
...
             raise MoonstreamHTTPException(status_code=500, internal_error=e)
 
-        if len(entries.results) == 0:
-            raise MoonstreamHTTPException(
-                status_code=403, detail="Query not approved yet."
-            )
     else:
         entries_results = cast(List[BugoutSearchResult], entries.results)
         query_id = entries_results[0].entry_url.split("/")[-1]
+        entry = entries_results[0]
```
Comment (kompotkot) [inline moonstreamapi/moonstreamapi/routes/queries.py:291]: I think this structure is wrong, what if we `try: entries = bc.search(` (line 252) failed, then we fetch line 277 and get entry, so our `entries is None` and we will fail when will try to get `entries.results` at line 289

**Review by kompotkot — APPROVED** [review]
lg

File: moonstreamapi/moonstreamapi/routes/queries.py (hunk 2 above)
Comment (Andrei-Dolgolev) [reply moonstreamapi/moonstreamapi/routes/queries.py:291]: `try: entries = bc.search( (line 252) failed` Give you raise MoonstreamHTTPException next code will be exacute only if search is finished.

Comment (kompotkot) [reply moonstreamapi/moonstreamapi/routes/queries.py:292]: But it return empty list, then `entries_results[0]` will fail

**Review by kompotkot — APPROVED** [review]
lg

**Pull request merged**